scipy/optimize/_basinhopping.py

```diff
@@ -310,7 +310,16 @@ def accept_reject(self, energy_new, energy_old):
         If new is higher than old, there is a chance it will be accepted,
         less likely for larger differences.
         """
-        w = math.exp(min(0, -float(energy_new - energy_old) * self.beta))
+        # energy_new, energy_old are sometimes scalars, sometimes arrays of
+        # length 1. Cast them all into scalars.
```

Comment:
This feels like something that should be fixed at the caller, not here

Reply:
Definitely. I tracked down the root cause to `_lbfgsb.setulb` returning an `array([val])` for the function value `f` in `scipy/optimize/lbfgsb.py`.

Reply:
Fixed now in LBFGS.